core/services/directrequest/delegate.go

```diff
@@ -54,13 +62,23 @@ func (d *Delegate) ServicesForSpec(spec job.Job) (services []job.Service, err er
 		return
 	}
 
+	minConfirmations := d.minRequiredOutgoingConfirmations
+	if concreteSpec.NumConfirmations > minConfirmations {
+		minConfirmations = concreteSpec.NumConfirmations
+	}
+
 	logListener := &listener{
-		logBroadcaster: d.logBroadcaster,
-		oracle:         oracle,
-		pipelineRunner: d.pipelineRunner,
-		db:             d.db,
-		pipelineORM:    d.pipelineORM,
-		spec:           *spec.PipelineSpec,
+		logBroadcaster:   d.logBroadcaster,
+		headBroadcaster:  d.headBroadcaster,
+		oracle:           oracle,
+		pipelineRunner:   d.pipelineRunner,
+		db:               d.db,
+		pipelineORM:      d.pipelineORM,
+		spec:             *spec.PipelineSpec,
+		mbLogs:           utils.NewMailbox(50),
```

Comment:
Probably ought to make this configurable, given how much we're experimenting with new chains

Reply:
Agree ^^. Also, do we log loudly if this is exceeded? It would be a pretty serious problem since we'd be dropping jobs.

Reply:
hm yeah - i added logging in this file on Deliver, and added a story to do it in other places https://app.clubhouse.io/chainlinklabs/story/7236/mailbox-improvements
Or do you think it should be done here? 

As for the config, i created MAILBOX_CAPACITY, but maybe you meant something more log-specific?